### PR TITLE
Dark mode with styled components

### PR DESCRIPTION
**Pull request by robcrock**
Here's a recap of what's working and what isn't: https://www.loom.com/share/56945b90f84e4b05913ab894a819ca4d

### CODE REVIEW

File: src/charts/AreaChart/AreaChart.tsx [hunk 1]
```diff
@@ -49,6 +71,7 @@ export default function AreaChart({
   legendLabel = '',
   chartType = 'area-chart',
   colorScheme = 'schemePurples',
+  tooltipVisible = false,
```
Comment (eric-mulhern) [inline src/charts/AreaChart/AreaChart.tsx]: Would be better if we made tooltipVisible default to true

File: src/charts/PieChart/PieChart.tsx [hunk 2]
```diff
@@ -14,7 +14,35 @@ import {
   EXTRA_LEGEND_MARGIN,
 } from '../../utils';
 
+import styled, { ThemeProvider } from 'styled-components';
+
+const LightTheme = {
```
Comment (eric-mulhern) [inline src/charts/PieChart/PieChart.tsx]: If these themes are the same for all charts, can we declare them in utils and import from there?

File: src/components/ColorLegend.tsx [hunk 3]
```diff
@@ -158,19 +175,19 @@ export const ColorLegend = ({
...
-    border: `1px solid #ddd`,
-    borderRadius: '4px',
-    // You can also use a fixed width and ommit the white-sapce.
-    whiteSpace: 'nowrap',
-    backgroundColor: '#ffffff',
-    fontFamily: 'Tahoma, Geneva, Verdana, sans-serif',
-    fontSize: '12px',
-    color: '#737373',
-  };
+  // const style: React.CSSProperties | undefined = {
```
Comment (eric-mulhern) [inline src/components/ColorLegend.tsx]: If not being used, should be deleted

File: src/components/TooltipContent.tsx [hunk 4]
```diff
@@ -0,0 +1,103 @@
+/* eslint-disable @typescript-eslint/restrict-plus-operands */
```
Comment (eric-mulhern) [inline src/components/TooltipContent.tsx:1]: Can these comments be cleaned up? Or do we need them for something later

File: types.ts [hunk 5]
```diff
@@ -52,9 +53,11 @@ export interface ScatterPlotProps<T> {
     | 'schemeGreens'
     | 'schemeBlues'
     | 'schemeSpectral';
+  tooltipVisible?: false | true;
```
Comment (eric-mulhern) [inline types.ts]: can refactor to `tooltipVisible?: boolean`

**Review by eric-mulhern — COMMENTED** [review]
• tooltipVisible should default to true
• On bar chart, tick text is always black:
Screen Shot 2022-01-31 at 16 01 52
• It seems that the legend and outer ring of the tooltip get their color from the 'theme' property of each chart component, while the center of the tooltip gets its color from the GlobalTheme. Can we set the center of the tooltip to always match the color from the 'theme' prop, so that we don't run into this?
Screen Shot 2022-01-31 at 15 57 55
Screen Shot 2022-01-31 at 16 05 20
• A few opportunities for refactoring, see comments
• Over all, looks really good!

File: src/charts/PieChart/PieChart.tsx (hunk 2 above)
Comment (robcrock) [reply src/charts/PieChart/PieChart.tsx]: Great recommendation! I was rushing 😅

File: src/components/ColorLegend.tsx (hunk 3 above)
Comment (robcrock) [reply src/components/ColorLegend.tsx]: Totally right

File: src/components/TooltipContent.tsx (hunk 4 above)
Comment (robcrock) [reply src/components/TooltipContent.tsx:1]: Those are actually functional comments.

File: types.ts (hunk 5 above)
Comment (robcrock) [reply types.ts]: true!

**Review by hi2dmitri — APPROVED** [review]
tooltip in bar chart is not displayed correctly. Changes from previous PR will be reverted

**Pull request merged**